semgrep_output_v0.atd

```diff
@@ -108,6 +108,20 @@ type core_match_extra
   <python decorator="dataclass(frozen=True)"> = {
   ?message: string option; (* rule.message (?) *)
   metavars: metavars;
+  (* Optional: present only for taint rule matches *)
+  ?taint_trace: taint_trace option;
+}
+
+type taint_trace
```

Comment:
Recommendation:

We may want to use this for other dataflow applications beyond taint. @kopecs had a suggestion to instead return an array of context information with arbitrary tags (perhaps with labels drawn from an enum).

In that case you might get something more like (in TS, b/c I'm not super familiar with ATD):

```typescript
type DataflowTag = "taint_source" | "taint_intermediate_var";
type DataflowContextItem = {
  tag: DataflowTag;
  location: Location;
}
type DataflowContext = array[DataflowContextItem];
```

Reply:
As we discussed elsewhere, your proposal has the downside that adding additional variants later is a breaking change. If consumers don't handle a default case, we have to make sure to upgrade them first, adding complexity. It also loses some of the structure that the current proposal provides.

I believe we could accomplish the same thing by leaving the current structure as-is, but giving it a more generic name. If we want to add new kinds of dataflow context items, we can simply add them to the record type alongside `source` and `intermediate_vars`, potentially as an optional field. If `source` or `intermediate_vars` isn't needed, we can just use an empty list. I'd also want to rename `source` to `taint_source` in that case.

I believe this will allow us to extend this for other purposes later, maintain the structure, and avoid any breaking changes. What do you think?

Reply:
sure sgtm